docs/getting-started/getting-started-with-gossip-ring.md

```diff
@@ -0,0 +1,186 @@
+```
+$ ./cortex -config.file docs/configuration/single-process-config-blocks-gossip-1.yaml
+$ ./cortex -config.file docs/configuration/single-process-config-blocks-gossip-2.yaml
+```
+
+## How it works
+
+The two instances we started earlier should be able to find each other via memberlist configuration (already present in the config files):
+
+```yaml
+memberlist:
+  # defaults to hostname
+  node_name: "Ingester 1"
+  bind_port: 7946
+  join_members:
+    - localhost:7947
+  abort_if_cluster_join_fails: false
+```
+
+This tells memberlist to listen on port 7946, and connect to localhost:7947, which is the second instance.
+Port numbers are reversed in the second configuration file.
+We also need to configure `node_name` and also ingester ID (`ingester.lifecycler.id` field), because default to hostname,
+but we are running both Cortex instances on the same host.
+
+To make sure that both ingesters generate unique tokens, we configure `join_after` and `observe_period` to 10 seconds.
+First option tells Cortex to wait 10 seconds before joining the ring.  This option is normally used to tell Cortex ingester
+how long to wait for a potential tokens and data transfer from leaving ingester, but we also use it here to increase
+the chance of finding other gossip peers. When Cortex joins the ring, it generates tokens and writes them to the ring.
+If multiple Cortex instances do this at the same time, they can generate conflicting tokens. This can be a problem
+when using gossiped ring (instances may simply not see each other yet), so we use `observe_period` to watch the ring for token conflicts.
+If conflict is detected, new tokens are generated instead of conflicting tokens, and observe period is restarted.
+If no conflict is detected within the observe period, ingester switches to ACTIVE state.
+
+We are able to observe ring state on [http://localhost:9109/ring](http://localhost:9109/ring) and [http://localhost:9209/ring](http://localhost:9209/ring).
+The two instances may see slightly different views (eg. different timestamps), but should converge to a common state soon, with both instances
+being ACTIVE and ready to receive samples.
+
+Download Prometheus and configure it to use our first Cortex instance for remote writes.
+
+```yaml
+remote_write:
+- url: http://localhost:9109/api/prom/push
+```
+
+After starting Prometheus, it will now start pushing data to Cortex. Distributor component in Cortex will
+distribute incoming samples between the two instances.
+
+To query that data, you can configure your Grafana instance to use http://localhost:9109/api/prom (first Cortex) as a Prometheus data source.
+
+## How to add another instance?
+
+To add another Cortex to the small cluster, copy `docs/configuration/single-process-config-blocks-gossip-1.yaml` to a new file,
+and make following modifications. We assume that third Cortex will run on the same machine again, so we change node name and ingester ID as well:
+
+```diff
```

Comment:
What do you think if we diff only the config that needs to be changed instead of copy-pasting the entire config? The main risk is that if we copy-paste the entire config, this will drift the first time we'll change the `.yaml` (and will happen).

Reply:
Good idea.